Tests: Emulate form blocks experiments in integration tests

aduth opened the pull request:
## What?

Updates "full content" integration tests to avoid explicitly registering form blocks, and instead sets up the environment that they would be registered as if the requisite experiment is enabled.

Related to #72978

## Why?

As discussed in #72978, the import subpaths for `@wordpress/block-library` as used in these tests are not [valid exports](https://github.com/WordPress/gutenberg/blob/48a6e8ae52d97441b0bbb593162124eb6c74dd46/packages/block-library/package.json#L27-L36) for the package, and only work because Jest presumably doesn't actually respect the exports API. This means it's likely to break in the future.

While we could update these imports to import from `build-module`, that means that the tests would only pass if the package were built ahead of time, which is not a requirement we should want to impose ([see related code](https://github.com/WordPress/gutenberg/blob/48a6e8ae52d97441b0bbb593162124eb6c74dd46/.github/workflows/unit-test.yml#L63-L65)).

## How?

Updates test to assign the `window` global that's evaluated when considering to register these blocks ([source](https://github.com/WordPress/gutenberg/blob/48a6e8ae52d97441b0bbb593162124eb6c74dd46/packages/block-library/src/index.js#L272)).

## Testing Instructions

`npm run test:unit test/integration/full-content/full-content.test.js`

aduth committed:
Tests: Emulate form blocks experiments in integration tests

jsnajdr commented on test/integration/full-content/full-content.test.js:
```diff
@@ -70,17 +64,11 @@ describe( 'full post content fixture', () => {
-			formSubmissionNotification,
-		] );
+		// in on the experimental settings page.
+		originalEnableFormBlocks = window.__experimentalEnableFormBlocks;
```
I'm a bit surprised that you have to restore the original `window` state. The `window` object should be synthetic, created anew for each test. By the `jest-environment-jsdom` package.

aduth replied:
I didn't actually test to verify this, but I did do some searching yesterday that concluded that that the window environment is **not** reset automatically between tests. I'll try to pull up some of those resources to confirm or counter that claim.

aduth replied:
Okay, after looking again, I believe the behavior is that the `window` is reset between test files, and shared within a test file (https://github.com/jestjs/jest/issues/1224#issuecomment-444791100). Since we're using this as an `beforeAll` / `afterAll`, I suppose it should be fine to set without restoring.

aduth replied:
Updated to remove in 3732e6af6e1

jsnajdr replied:
There are different policies for modules, mocks and the environment (`window` etc.). Mocks and modules are shared by the entire test file, but the environment should be new for each individual test.

jsnajdr reviewed (approved):
Yes, this is a good idea, turns out we don't need the granular `block-library` exports in this test.

aduth committed:
Avoid resetting window global for experimental blocks

The expected behavior for Jest + JSDOM should be that individual test files have an isolated window global

See: https://github.com/WordPress/gutenberg/pull/73705/files#r2585370933
